external/fv3fit/fv3fit/emulation/layers/architecture.py

```diff
@@ -429,11 +442,11 @@ def __init__(
         super().__init__(**kwargs)
 
         self.arch = _get_arch_layer(key, net_kwargs)
-        self.input_combiner = _get_combine_layer(key)
-        self.outputs = _get_output_layer(key, feature_lengths)
+        self.input_layer = self.arch.input_layer
+        self.outputs = self.arch.get_output_layer(feature_lengths)
 
     def call(self, tensors: Mapping[str, tf.Tensor]) -> Mapping[str, tf.Tensor]:
-        combined = self.input_combiner(tensors)
+        combined = self.input_layer(tensors)
```

Comment:
Not in this PR, but thought for later: the _HiddenArchitecture class could be deleted if you add a method to the layers in this file which calls their `input_layer`, current `__call__`, and `outputs` methods. This would improve useability a bit (you can inspect the return type of `ArcitectureConfig.build`), and give a better sense when reading the layer classes how they're actually used (by calling pre, compute, and post steps).

The documentation e.g. for RNNOutput describes that the class combines, RNN's, then connected outputs, so it confused me when the method doing this doesn't exist on that class. Particularly, I was trying to see how the classes use their input_layer and outputs methods. Eventually it made sense when I read more of the file.

Reply:
I ended up simplifying this class by pulling the constructor logic up the call stack. I think that `_HiddenArchitecture` is useful for composing these layers in flexible combinations. Ultimately it is pretty similar to `tf.keras.Sequential`, but it allows dictionary inputs and outputs.

Reply:
The class would be useful for flexible combinations, if that was a need, but I see below that for a given arch, you always use the same input and output layer classes. If an RNNOutput is kind of needed for an RNNBlock arch, it can be helpful to see that codified in the RNNBlock class.

This is a fairly small difference between two good solutions.